### PR TITLE
Add workload7 to explain pipeline

### PR DESCRIPTION
**Pull request by pobbatihari**
This PR includes YAML modifications to facilitate a 3TB explain pipeline and configures segments based on workloads

### CODE REVIEW

File: concourse/scripts/build_gpdb.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def main():
     parser = optparse.OptionParser()
     parser.add_option("--mode", choices=['orca', 'planner'])
     parser.add_option("--output_dir", dest="output_dir", default=INSTALL_DIR)
+    parser.add_option("--num_segments", dest="num_segments", help="number of segments")
```
Comment (chrishajas) [inline concourse/scripts/build_gpdb.py:47]: Can we add a default of 32 in case this is used by other scripts/repos

**Review by chrishajas — APPROVED** [review]
Looks good after adding the default

File: concourse/scripts/build_gpdb.py (hunk 1 above)
Comment (pobbatihari) [reply concourse/scripts/build_gpdb.py:47]: sure

File: concourse/scripts/builds/GpBuild.py [hunk 2]
```diff
@@ -79,8 +79,11 @@ def run_explain_test_suite(self, dbexists):
         # set gucs if any were specified
         status = self._run_cmd("source gpdb_src/gpAux/gpdemo/gpdemo-env.sh && cat gporca-commits-to-test/optional_gucs.txt >> $COORDINATOR_DATA_DIRECTORY/postgresql.conf", None)
         fail_on_error(status)
-        # use 32 segments for explain tests (this number is fairly arbitrary, it might be better to make this dependent on the workload?)
-        status = self._run_cmd("source gpdb_src/gpAux/gpdemo/gpdemo-env.sh && echo 'optimizer_segments=32\ngp_segments_for_planner=32' >> $COORDINATOR_DATA_DIRECTORY/postgresql.conf", None)
+
+        command = "source gpdb_src/gpAux/gpdemo/gpdemo-env.sh && echo 'optimizer_segments={0}\ngp_segments_for_planner={0}' >> $COORDINATOR_DATA_DIRECTORY/postgresql.conf".format(num_segments)
+        print("running command: " + command)
```
Comment (gpopt) [inline concourse/scripts/builds/GpBuild.py:84]: Running instead of running

File: src/backend/gporca/concourse/test_explain_pipeline.yml [hunk 3]
```diff
@@ -848,3 +882,100 @@ jobs:
...
+            disable_git_lfs: true
+            depth: 1
+        - get: gpdb7-rhel8-test
+    - task: run_explain_suite_baseline
+      file: gpdb_src/concourse/tasks/run_explain_suite.yml
+      image: gpdb7-rhel8-test
+      params:
+        MODE: orca
+        WORKLOAD: ((workload7))
+        NUM_SEGMENTS: 24
```
Comment (gpopt) [inline src/backend/gporca/concourse/test_explain_pipeline.yml:949]: Is the value of 24 intended?

Comment (pobbatihari) [reply src/backend/gporca/concourse/test_explain_pipeline.yml:949]: Since 3TB setup  has 24 segments, I've configured it accordingly as discussed

**Review by DevChattopadhyay — APPROVED** [review]
The changes look good to me.

**Review by gpopt — APPROVED** [review]
+1

**Pull request merged**